vainfo/vainfo.c

```diff
@@ -84,19 +91,421 @@ parse_args(const char *name, int argc, char **argv)
     }
 }
 
+static int show_config_attributes(VADisplay va_dpy, VAProfile profile, VAEntrypoint entrypoint)
+{
+  struct str_format
+  {
+    int format;
+    char *name;
+  };
+
+  VAStatus va_status;
+  int max_num_attributes, i, n;
+  VAConfigAttrib * attrib_list = NULL;
+  int ret_val = 0;
+
+  max_num_attributes = vaMaxNumConfigAttributes(va_dpy);
+  attrib_list = malloc (max_num_attributes * sizeof (VAConfigAttrib));
+  for (i = 0; i < max_num_attributes; i++) {
+      attrib_list[i].type = i;
```

Comment:
To pass list of [attrib](https://github.com/intel/libva/blob/master/va/va.h#L793) structure, to be filled, to vaGetConfigAttributes() we need to init _type_ field at first: 
```
typedef struct _VAConfigAttrib {
    VAConfigAttribType type;
    uint32_t value; /* OR'd flags (bits) for this attribute */
} VAConfigAttrib;

```
where VAConfigAttribType is enum starting from 0.

Reply:
vaMaxNumConfigAttributes returns number of attributes supported by driver implementation. It does not mean that implementation supports all of them. For example, it VAAPI might have 10 attributes exposed, but driver can support only 9 attributes missing 2nd one for example. So, setting config types as you do, does not actually makes sense.

Reply:
I suppose you should use vaQueryConfigAttributes in stead of vaGetConfigAttributes,  vaQueryConfigAttributes will list all supported attributes, vaGetConfigAttributes will show what you care about. as @dvrogozh mentioned, vaGetConfigAttributes just return several attributes in the enum list. it maybe not what you want

Reply:
The intention is to get **all** supported config attributes **per  a particular component** (defined by pair VAProfile/vaEntrypoint). While vaGetConfigAttributes allows to do it **statically** (i mean without additional parametrization),  vaQueryConfigAttributes requires a **temporal** config. This config should be creates using _some harcoded_ values which I aim to avoid.

Reply:
Hi @XinfengZhang ,

Did I address your concern?